### ISSUE TITLE
'Failed to make configuration' while building osxfuse on macOS 10.14

### ISSUE THREAD

**Issue by mixtly87** [osxfuse/osxfuse#562]
I'm facing following issue while trying to build the osxfuse:

```
$ ./build.sh -t distribution -v 5
...
T:library            | Clean target
T:library            | Source target library
T:library            | Invoke action clean
T:library            | BUILD_TARGET_OPTION_ARCHITECTURES=([0]=x86_64)
T:library            | BUILD_TARGET_OPTION_BUILD_CONFIGURATION=Release
T:library            | BUILD_TARGET_OPTION_BUILD_SETTINGS=()
T:library            | BUILD_TARGET_OPTION_CODE_SIGN_IDENTITY=''
T:library            | BUILD_TARGET_OPTION_DEBUG_DIRECTORY=''
T:library            | BUILD_TARGET_OPTION_DEPLOYMENT_TARGET=10.14
T:library            | BUILD_TARGET_OPTION_GROUP=''
T:library            | BUILD_TARGET_OPTION_MACROS=()
T:library            | BUILD_TARGET_OPTION_OWNER=''
T:library            | BUILD_TARGET_OPTION_PREFIX=/usr/local
T:library            | BUILD_TARGET_OPTION_PRODUCT_SIGN_IDENTITY=''
T:library            | BUILD_TARGET_OPTION_PROVISIONING_PROFILE_SPECIFIER=''
T:library            | BUILD_TARGET_OPTION_ROOT=0
T:library            | BUILD_TARGET_OPTION_SDK=10.14
T:library            | BUILD_TARGET_OPTION_XCODE=10.1
T:library            | Removing target build directory '/tmp/osxfuse/library'
T:library            | Completed action clean
T:library            | Build target for macOS 10.14
Running libtoolize...
Running autoreconf...
autom4te: need GNU m4 1.4 or later: /opt/local/bin/gm4
aclocal: error: echo failed with exit status: 1
autoreconf: aclocal failed with exit status: 1
T:library            | Failed to make configuration
Terminated: 15
Received signal: SIGTERM
```

Did anyone encounter this or does anyone know how to fix this?

**Comment by bfleischer** [comment]
Did you install all required tools (autoconf automake libtool gettext)? I'm still using 10.13 to build FUSE releases.

**Comment by mixtly87** [comment]
Hm, looks like I didn't have all tools installed. Anyways, I installed them and now the error is:
```
T:library            | Clean target
T:library            | Source target library
T:library            | Invoke action clean
T:library            | BUILD_TARGET_OPTION_ARCHITECTURES=([0]=x86_64)
T:library            | BUILD_TARGET_OPTION_BUILD_CONFIGURATION=Release
T:library            | BUILD_TARGET_OPTION_BUILD_SETTINGS=()
T:library            | BUILD_TARGET_OPTION_CODE_SIGN_IDENTITY=''
T:library            | BUILD_TARGET_OPTION_DEBUG_DIRECTORY=''
T:library            | BUILD_TARGET_OPTION_DEPLOYMENT_TARGET=10.14
T:library            | BUILD_TARGET_OPTION_GROUP=''
T:library            | BUILD_TARGET_OPTION_MACROS=()
T:library            | BUILD_TARGET_OPTION_OWNER=''
T:library            | BUILD_TARGET_OPTION_PREFIX=/usr/local
T:library            | BUILD_TARGET_OPTION_PRODUCT_SIGN_IDENTITY=''
T:library            | BUILD_TARGET_OPTION_PROVISIONING_PROFILE_SPECIFIER=''
T:library            | BUILD_TARGET_OPTION_ROOT=0
T:library            | BUILD_TARGET_OPTION_SDK=10.14
T:library            | BUILD_TARGET_OPTION_XCODE=10.1
T:library            | Removing target build directory '/tmp/osxfuse/library'
T:library            | Completed action clean
T:library            | Build target for macOS 10.14
Running libtoolize...
Running autoreconf...
configure.ac:77: warning: macro 'AM_ICONV' not found in library
configure.ac:77: warning: macro 'AM_ICONV' not found in library
configure.ac:77: error: possibly undefined macro: AM_ICONV
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/local/Cellar/autoconf/2.69/bin/autoconf failed with exit status: 1
T:library            | Failed to make configuration
Terminated: 15
Received signal: SIGTERM
```

However, I can also successfully build on 10.13, but when I run Distribution.pkg I get an Alert that 'The FUSE for macOS installation package is not compatible with this version of macOS.'

The very same installer works OK on macOS Mojave 10.14.2 however.

So now we have two issues:
1. Why build.sh fails on macOS Mojave?
2. Why Distribution.pkg which is build on 10.13 is not compatible on macOS 10.13 but it is on macOS 10.14?

**Comment by bfleischer** [comment]
Let's start with the build error. Please read the README.md. When using Homebrew you need to run `brew link --force gettext` otherwise its macros are not found.

When building the FUSE kernel extension you need the specific SDK for the version of macOS that the kernel extension is supposed to run under. This means, to build FUSE for macOS 10.13 you need an Xcode version that includes the 10.13 SDK. The last version of Xcode to include the 10.13 SDK is 9.4.1. I'm assuming you are using Xcode 10, correct? Then you need to downgrade Xcode or install Xcode 9 in addition to Xcode 10.

**Comment by mixtly87** [comment]
1. Fixed. osxfuse now successfully builds on macOS Mojave. Looks like the problem was that I didn't follow the README instructions carefully.

2. Thanks for the tip! I'll try installing Xcode 9 alongside Xcode 10 (if it's possible at all) and check back the results.

**Comment by bfleischer** [comment]
Yes, you can certainly install Xcode 9 alongside Xcode 10. By the way, official releases are built using Xcode 3.2.6 (patched), Xcode 6.4, Xcode 7.3.1, and Xcode 10.1, all on the same machine running macOS 10.13.
